singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java

```diff
@@ -241,7 +247,8 @@ public long processLogStream() throws LogStreamProcessorException {
 
   /**
    * If the decider is not set, this method will return true.
-   * If a decider is set, only return false when the decider's value is 0.
+   * If a decider is set, only return false when the decider's value is 0 and disable decider's
+   * (if exists) value is 100.
```

Comment:
Is it "and" or "or"? I.e. do both conditions need to hold? Or either?

Reply:
Yes both conditions need to hold if an applicable disable decider exists for the host

Reply:
Am I missing something here?

Logging is allowed (return true) if
-  logstream decider value is non-zero, and
- disable deccider value is not 100.

Then, logging is disallowed (return false) if (logical negation of "a and b" is "~a or ~b")
- logstream decider value is 0, or
- disable decider value is 100

Reply:
Oh sorry yes I misread your comment, I mean to convey that disable decider will only be considered if logsteam decider is non-zero. Let me change this comment